src/components/SelectionList/BaseSelectionList.js

```diff
@@ -152,31 +152,31 @@ function BaseSelectionList({
      * @param {Number} index - the index of the item to scroll to
      * @param {Boolean} animated - whether to animate the scroll
      */
-    const scrollToIndex = useCallback((index, animated = true) => {
-        const item = flattenedSections.allOptions[index];
+    const scrollToIndex = useCallback(
+        (index, animated = true) => {
+            const item = flattenedSections.allOptions[index];
 
-        if (!listRef.current || !item) {
-            return;
-        }
+            if (!listRef.current || !item) {
+                return;
+            }
 
-        const itemIndex = item.index;
-        const sectionIndex = item.sectionIndex;
+            const itemIndex = item.index;
+            const sectionIndex = item.sectionIndex;
 
-        // Note: react-native's SectionList automatically strips out any empty sections.
-        // So we need to reduce the sectionIndex to remove any empty sections in front of the one we're trying to scroll to.
-        // Otherwise, it will cause an index-out-of-bounds error and crash the app.
-        let adjustedSectionIndex = sectionIndex;
-        for (let i = 0; i < sectionIndex; i++) {
-            if (_.isEmpty(lodashGet(sections, `[${i}].data`))) {
-                adjustedSectionIndex--;
+            // Note: react-native's SectionList automatically strips out any empty sections.
+            // So we need to reduce the sectionIndex to remove any empty sections in front of the one we're trying to scroll to.
+            // Otherwise, it will cause an index-out-of-bounds error and crash the app.
+            let adjustedSectionIndex = sectionIndex;
+            for (let i = 0; i < sectionIndex; i++) {
+                if (_.isEmpty(lodashGet(sections, `[${i}].data`))) {
+                    adjustedSectionIndex--;
+                }
             }
-        }
-
-        listRef.current.scrollToLocation({sectionIndex: adjustedSectionIndex, itemIndex, animated, viewOffset: variables.contentHeaderHeight});
 
-        // If we don't disable dependencies here, we would need to make sure that the `sections` prop is stable in every usage of this component.
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
+            listRef.current.scrollToLocation({sectionIndex: adjustedSectionIndex, itemIndex, animated, viewOffset: variables.contentHeaderHeight});
+        },
+        [flattenedSections.allOptions, sections],
```

Comment:
@rushatgabhane Getting below error if we pass only `flattenedSections.allOptions` to dependencies, so to solve that warning, I've added the sections to the dependencies; not sure if this is a good idea as this line (`we would need to make sure that the `**sections**` prop is stable in every usage of this component`)
 suggests we should check if sections are stable in every usage, right now `SelectionList` is being used at 14 pages. 
 <img width="1284" alt="Screenshot 2023-10-18 at 4 43 47 PM" src="https://github.com/Expensify/App/assets/35371050/ad00d95d-9084-4ba5-b93b-c471cbc49f2c">

Reply:
@rushatgabhane Thanks for the suggestion. It would also work. I have added disable exhaustive-deps and pushed code.